### PR TITLE
fix: Update the way system bars info is parsed

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
It looks like dumpsys output is changing between different Android versions, so I've updated the parsing rules to make it work on Android 12+. Also refactored the code to make it a bit more easier to modify and update.

Addresses https://github.com/appium/appium/issues/16390

I've only tested on Android 11 and 12 emulators. I assume more testing is necessary on older system versions though.

### CODE REVIEW

**Review by jlipps — APPROVED** [review]
Code looks good if it works!

**Comment by KazuCocoa** [comment]
Android 6 (tablet) was good, both landscape and portrait. Android 10 was as well.

**Comment by KazuCocoa** [comment]
a unity game demo app also worked, each bar was invisible and visible in both cases.

File: lib/commands/system-bars.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    log.info(`No windows have been found whose title matches to ` +
+      `'${namePrefix}'. Assuming it is invisible. ` +
+      `Only the following windows are available: ${_.keys(windows)}`);
+    result[window] = _.cloneDeep(DEFAULT_WINDOW_PROPERTIES);
+  }
+  return result;
+}
+
+commands.getSystemBars = async function getSystemBars () {
+  const out = await this.adb.shell(['dumpsys', 'window', 'windows']);
```
Comment (KazuCocoa) [inline lib/commands/system-bars.js:120]: need try/catch? e.g. when the device is disconnected at this time

**Review by KazuCocoa — APPROVED** [review]

File: lib/commands/system-bars.js (hunk 1 above)
Comment (mykola-mokhnach) [reply lib/commands/system-bars.js]: added

**Comment by mykola-mokhnach** [comment]
Thanks for checking it @KazuCocoa

**Pull request merged**